src/galileo/handlers/openai_agents.py

```diff
@@ -0,0 +1,333 @@
+
+from agents import TracingProcessor, tracing
+
+from galileo.decorator import galileo_context
+from galileo.logger import GalileoLogger
+from galileo.utils.serialization import convert_to_string_dict, serialize_to_str
+
+_logger = logging.getLogger("galileo.handlers.agents")
+
+
+def _get_span_name(span: tracing.Span[Any]) -> str:
+    """Get name for a given OpenAI Agent span."""
+    if name := getattr(span.span_data, "name", None):
+        return str(name)
+    elif isinstance(span.span_data, tracing.GenerationSpanData):
+        return "Generation"
+    elif isinstance(span.span_data, tracing.ResponseSpanData):
+        return "Response"
+    elif isinstance(span.span_data, tracing.HandoffSpanData):
+        return "Handoff"
+    elif isinstance(span.span_data, tracing.FunctionSpanData):
+        return "FunctionCall"  # More descriptive than 'task' potentially
+    elif isinstance(span.span_data, tracing.GuardrailSpanData):
+        return "Guardrail"
+    elif isinstance(span.span_data, tracing.AgentSpanData):
+        return "AgentStep"  # More descriptive than 'task' potentially
+    elif isinstance(span.span_data, tracing.CustomSpanData):
+        return "CustomStep"
+    else:
+        # Use the type if available, otherwise fallback
+        type_name = span.span_data.type or "UnknownStep"
+        return type_name.capitalize()
+
+
+class GalileoTracingProcessor(TracingProcessor):  # pyright: ignore[reportGeneralTypeIssues]
+    """
+    TracingProcessor that logs OpenAI Agent traces and spans to Galileo.
+
+    This processor captures different types of spans from OpenAI Agents and logs them
+    using the appropriate Galileo span types (workflow, llm, tool, etc.).
+    """
+
+    def __init__(self, galileo_logger: Optional[GalileoLogger] = None, flush_on_chain_end: bool = True) -> None:
+        self._galileo_logger: GalileoLogger = galileo_logger or galileo_context.get_logger_instance()
+        self._trace_started: set[str] = set()  # Track active traces
+        self._flush_on_chain_end: bool = flush_on_chain_end
+
+    def on_trace_start(self, trace: tracing.Trace) -> None:
+        """Called when a trace starts. Starts a Galileo trace."""
+        try:
+            # Use trace name as input, similar to Langchain example's root node
+            self._galileo_logger.start_trace(input=trace.name or "Unnamed Agent Trace")
+            self._trace_started.add(trace.trace_id)
+            _logger.debug(f"Started Galileo trace for agent trace_id: {trace.trace_id}")
+        except Exception as e:
+            _logger.error(f"Galileo: Error in on_trace_start for trace_id {trace.trace_id}: {e}", exc_info=True)
+
+    def on_trace_end(self, trace: tracing.Trace) -> None:
+        """Called when a trace ends. Concludes the Galileo trace."""
+        if trace.trace_id not in self._trace_started:
+            _logger.warning(f"Galileo: Received on_trace_end for unknown trace_id: {trace.trace_id}")
+            return
+        try:
```

Comment:
you can move it to smaller function, then you can use decorator as well instead.